CONTRIBUTING.rst

```diff
@@ -464,9 +487,97 @@ Step 5: Pass PR Review
+  the right to veto any code modification according to the `Apache Software Foundation rules <https://www.apache.org/foundation/voting.html#votes-on-code-modification>`_.
+
+* ``Approval`` - this is given by a maintainer after the code has been reviewed and the maintainer agrees that
+  it is a good idea to merge it. There might still be some unresolved conversations, requests and questions on
+  such PR and you are expected to resolve them before the PR is merged. But the ``Approval`` status is a sign
+  of trust from the maintainer who gave the approval that they think the PR is good enough as long as their
+  comments will be resolved and they put the trust in the hands of the author and - possibly - other
+  maintainers who will merge the request that they can do that without follow-up re-review and verification.
+
+
+You need to have ``Approval`` of at least one committer (if you are committer yourself, it has to be
 another committer). Ideally you should have 2 or more committers reviewing the code that touches
-the core of Airflow.
+the core of Airflow - we do not have enforcement about ``2+`` reviewers required for Core of Airflow,
+but Maintainers will generally ask in the PR if they think second review is needed.
+
+Your PR can be merged by the reviewer or another maintainer who will see that the
+PR is approved, all conversations are resolved and the code looks good. The criteria for PR
+being merge-able are:
+
+* ``green status for static checks and tests``
+* ``conversations resolved``
+* ``Approval from 1 (or more for core changes) maintainers``
+* no unresolved ``Request changes``
+
+Once you reach the status, you do not need to do anything to get the PR merged. One of the maintainers
+will merge such PR. However if you see that for a few days such PR is not merged, do not hesitate to comment
+on your PR and mention that you think it is ready to be merged. Also it's a good practice to rebase your PR
+to latest ``main``, because there could be other changes merged in the meantime that might cause conflicts or
+fail tests or static checks, so by rebasing a PR that has been build few days ago you make sure that it
+still passes the tests and static checks today.
+
+
+.. note:: |experimental|
+
+   In December 2023 we enabled - experimentally - the requirement to resolve all the open conversations in a
+   PR in order to make it merge-able. You will see in the status of the PR that it needs to have all the
+   conversations resolved before it can be merged.
+
+   This is an experiment and we will evaluate by the end of January 2024. If it turns out to be a good idea,
+   we will keep it enabled in the future.
+
+   The goal of this experiment is to make it easier to see when there are some conversations that are not
+   resolved for everyone involved in the PR - author, reviewers and maintainers who try to figure out if
+   the PR is ready to merge and ... merge it. The goal is also to use conversations more as a "soft" way
```

Comment:
The ... is intentional?

Reply:
Yeah. I use ellipsis too much as well. Changed to more explicit ` - eventually - `